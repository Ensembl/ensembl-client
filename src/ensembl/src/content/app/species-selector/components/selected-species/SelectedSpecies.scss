@import 'src/styles/common';

$selectedSpeciesBorderRadius: 20px;
$selectedSpeciesPadding: 6px 20px;
$deleteButtonPadding: 8px 20px 8px 28px;

.selectedSpecies {
  position: relative;
  display: inline-block;
  padding: $selectedSpeciesPadding;
  background-color: $ens-black;
  border-radius: $selectedSpeciesBorderRadius;
  line-height: 1;
  cursor: pointer;
  user-select: none;

  &:hover {
    background-color: transparent;
  }

  &:not(:last-child) {
    margin-right: 1em;
  }

  .name {
    font-size: 14px;
    color: white;
    font-weight: $bold;
    margin-right: 0.5rem;
  }

  .assembly {
    font-size: 11px;
    color: $white;
  }
}

.selectedSpeciesDisabled {
  background-color: $ens-grey;
}

.selectedSpeciesOverlay {
  position: absolute;
  left: 0;
  top: 0;
  z-index: 1;
  width: 100%;
  height: 100%;
  line-height: 1;
  background: white;
  border: 1px solid $ens-blue;
  border-radius: $selectedSpeciesBorderRadius;
  cursor: default;

  .overlayText,
  .closeButtonContainer {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    cursor: pointer;
  }

  .overlayText {
    padding: $selectedSpeciesPadding;
    left: 0;
    color: $ens-blue;
  }

  .closeButtonContainer {
    display: flex;
    padding: $deleteButtonPadding;
    align-items: center;
    right: -0.5em;
  }
}
