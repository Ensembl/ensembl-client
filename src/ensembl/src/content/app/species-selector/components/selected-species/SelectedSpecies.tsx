import React, { FunctionComponent, useState } from 'react';
import classNames from 'classnames';
import noop from 'lodash/noop';

import CloseButton from 'src/shared/close-button/CloseButton';

import styles from './SelectedSpecies.scss';

import { CommittedItem } from 'src/content/app/species-selector/types/species-search';

type Props = {
  species: CommittedItem;
  onToggleUse: (genomeId: string) => void;
  onRemove: (genomeId: string) => void;
};

const SelectedSpecies: FunctionComponent<Props> = (props) => {
  const [hovered, setHovered] = useState(false);
  const setHoveredState = () => setHovered(true);
  const setUnhoveredState = () => setHovered(false);

  const {
    common_name,
    scientific_name,
    assembly_name,
    isEnabled
  } = props.species;

  const displayName = common_name || scientific_name;

  const className = classNames(styles.selectedSpecies, {
    [styles.selectedSpeciesDisabled]: !isEnabled
  });

  return (
    <div
      className={className}
      onMouseOver={setHoveredState}
      onMouseLeave={setUnhoveredState}
    >
      <span className={styles.name}>{displayName}</span>
      <span className={styles.assembly}>{assembly_name}</span>
      {hovered && <SelectedSpeciesOverlay {...props} />}
    </div>
  );
};

const SelectedSpeciesOverlay: FunctionComponent<Props> = (props) => {
  const { genome_id, isEnabled } = props.species;

  const text = isEnabled ? 'Do not use' : 'Use';

  const handleClick = () => {
    props.onToggleUse(genome_id);
  };
  const removeSpecies = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    props.onRemove(genome_id);
  };

  return (
    <div className={styles.selectedSpeciesOverlay}>
      <span className={styles.overlayText} onClick={handleClick}>
        {text}
      </span>
      <div className={styles.closeButtonContainer} onClick={removeSpecies}>
        <CloseButton inverted={true} onClick={noop} />
      </div>
    </div>
  );
};

export default SelectedSpecies;
