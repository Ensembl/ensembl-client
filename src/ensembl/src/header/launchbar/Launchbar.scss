@import 'src/styles/common';

.launchbar {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid $ens-grey;
  border-top: 1px solid $ens-grey;
  padding: 8px 6px;
}

.categoriesWrapper {
  display: flex;
  flex-wrap: nowrap;
  overflow-x: auto;

  &::-webkit-scrollbar {
    display: none;
  }
}

.categories {
  display: flex;
  flex: 0 0 auto;
}

.category {
  display: flex;
  padding: 0 15px;

  &:first-child {
    padding-left: 2px;
  }

  &:not(:last-child) {
    border-right: 1px solid $ens-grey;
  }
}

.launchbarButton {
  display: inline-block;
  height: 30px;
  width: 30px;
  border: 1px solid $ens-blue;
  margin: 0 12px;

  svg {
    width: 100%;
    height: 100%;
    transform: scale(1.1);
    fill: $ens-white;
    background-color: $ens-blue;
  }

  img {
    max-width: 100%;
    max-height: 100%;
  }
}

.launchbarButtonSelected {
  border: 1px solid $ens-black;

  svg {
    fill: $ens-white;
    background-color: $ens-black;
  }
}

.launchbarButtonDisabled {
  border: 1px solid $ens-grey;

  svg {
    fill: $ens-white;
    background-color: $ens-grey;
  }
}

.about {
  display: flex;
  align-items: center;
  margin-left: 20px;

  .aboutText {
    font-size: 12px;
    margin-right: 3px;
    margin-top: 2px;
  }

  .launchbarButton {
    border: none;

    &:last-child {
      margin-right: 18px;
    }
  }
}
