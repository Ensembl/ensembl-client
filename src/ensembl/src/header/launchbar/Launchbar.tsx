import React, { FunctionComponent, ReactNode } from 'react';
import SlideDown from 'react-slidedown';

import {
  LaunchbarCategory,
  launchbarConfig,
  LaunchbarApp
} from './launchbarConfig';

import LaunchbarIcon from './LaunchbarIcon';

import styles from './Launchbar.scss';

type LaunchbarProps = {
  currentApp: string;
  launchbarExpanded: boolean;
};

export const getCategoryClass = (separator: boolean): string => {
  return separator ? 'border' : '';
};

const Launchbar: FunctionComponent<LaunchbarProps> = (
  props: LaunchbarProps
) => {
  const LaunchbarChildren: ReactNode = (
    <div className={styles.launchbar}>
      <div className={styles.categoriesWrapper}>
        <div className={styles.categories}>
          {launchbarConfig.categories.map((category: LaunchbarCategory) => (
            <dl
              className={styles[getCategoryClass(category.separator)]}
              key={category.name}
            >
              {category.apps.map((app: LaunchbarApp) => (
                <dt key={app.name} className={`${app.name}Icon`}>
                  <LaunchbarIcon app={app} currentApp={props.currentApp} />
                </dt>
              ))}
            </dl>
          ))}
        </div>
      </div>
      <div className={styles.about}>
        <dl>
          <dt className="aboutIcon">
            <LaunchbarIcon
              app={launchbarConfig.about}
              currentApp={props.currentApp}
            />
          </dt>
        </dl>
      </div>
    </div>
  );

  return (
    <SlideDown transitionOnAppear={false}>
      {props.launchbarExpanded ? LaunchbarChildren : null}
    </SlideDown>
  );
};

export default Launchbar;
