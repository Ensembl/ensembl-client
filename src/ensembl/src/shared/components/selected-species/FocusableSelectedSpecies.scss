@import 'src/styles/common';
@import './selected-species-common';

.species {
  @include selected-species-common;
  border: 1px solid $blue;
  cursor: pointer;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  // flex-grow: 0;
  // flex-shrink: 1;
  // flex-basis: auto;
  // transition: flex-shrink 200ms;
}

.name,
.assembly {
  color: $black;
}

.speciesActive {
  background-color: $black;
  border: none;
  cursor: default;
}

.nameActive,
.assemblyActive {
  color: $white;
}
