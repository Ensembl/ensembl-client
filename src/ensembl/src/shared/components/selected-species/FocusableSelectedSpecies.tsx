import React from 'react';
import classNames from 'classnames';

import SelectedSpeciesDisplayName from './SelectedSpeciesDisplayName';

import styles from './FocusableSelectedSpecies.scss';

import { CommittedItem } from 'src/content/app/species-selector/types/species-search';

export type Props = {
  species: CommittedItem;
  isActive: boolean;
  onClick: (genomeId: string) => void;
  onMouseEnter?: () => void;
  onMouseLeave?: () => void;
  className?: string;
};

const SpeciesTab = (props: Props) => {
  const handleMouseEnter = () => {
    props.onMouseEnter && props.onMouseEnter();
  };

  const handleMouseLeave = () => {
    props.onMouseLeave && props.onMouseLeave();
  };

  const handleClick = () => {
    if (!props.isActive) {
      props.onClick(props.species.genome_id);
    }
  };

  const className = classNames(styles.species, props.className, {
    [styles.speciesActive]: props.isActive
  });
  const nameClass = props.isActive ? styles.nameActive : styles.name;
  const assemblyClass = props.isActive
    ? styles.assemblyActive
    : styles.assembly;

  return (
    <div
      className={className}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      onClick={handleClick}
    >
      <SelectedSpeciesDisplayName
        species={props.species}
        nameClassName={nameClass}
        assemblyClassName={assemblyClass}
      />
    </div>
  );
};

export default SpeciesTab;
