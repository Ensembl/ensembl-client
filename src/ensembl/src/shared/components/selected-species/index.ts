export { default as SimpleSelectedSpecies } from './SimpleSelectedSpecies';
export {
  default as FocusableSelectedSpecies
} from './FocusableSelectedSpecies';
