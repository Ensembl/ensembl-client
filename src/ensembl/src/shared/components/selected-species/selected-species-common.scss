@import 'src/styles/common';

$selectedSpeciesBorderRadius: 20px;
$selectedSpeciesPadding: 6px 20px;
$deleteButtonPadding: 8px 20px 8px 28px;

@mixin selected-species-common {
  display: inline-block;
  padding: $selectedSpeciesPadding;
  border-radius: $selectedSpeciesBorderRadius;
  line-height: 1;
  user-select: none;
  margin-bottom: 4px;

  &:not(:last-child) {
    margin-right: 12px;
  }
}

.name,
.assembly {
  color: $white;
}

.name {
  font-size: 14px;
  font-weight: $bold;
  margin-right: 0.5rem;
}

.assembly {
  font-size: 11px;
}
