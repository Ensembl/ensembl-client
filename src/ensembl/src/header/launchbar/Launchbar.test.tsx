import React from 'react';
import { mount } from 'enzyme';

import Launchbar from './Launchbar';

import { createSelectedSpecies } from 'tests/fixtures/selected-species';

jest.mock('./LaunchbarButton', () => () => <div>Launchbar Button</div>);

const defaultProps = {
  launchbarExpanded: true,
  committedSpecies: []
};

describe('<Launchbar />', () => {
  it('disables Genome Browser button when there are no committed species', () => {
    const wrapper = mount(<Launchbar {...defaultProps} />);
    const genomeBrowserButton = wrapper.findWhere(
      (wrapper) => wrapper.prop('app') === 'browser'
    );

    expect(genomeBrowserButton.prop('enabled')).toBe(false);
  });

  it('enables Genome Browser button when there are committed species', () => {
    const props = {
      ...defaultProps,
      committedSpecies: [createSelectedSpecies()]
    };
    const wrapper = mount(<Launchbar {...props} />);
    const genomeBrowserButton = wrapper.findWhere(
      (wrapper) => wrapper.prop('app') === 'browser'
    );

    expect(genomeBrowserButton.prop('enabled')).toBe(true);
  });
});
