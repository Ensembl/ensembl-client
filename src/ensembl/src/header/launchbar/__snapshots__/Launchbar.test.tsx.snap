// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Launchbar /> renders correctly 1`] = `
<ForwardRef
  transitionOnAppear={false}
>
  <div
    className="launchbar"
  >
    <div
      className="categoriesWrapper"
    >
      <div
        className="categories"
      >
        <dl
          className="border"
          key="search"
        >
          <dt
            className="global-searchIcon"
            key="global-search"
          >
            <React.Memo
              app={
                Object {
                  "description": "global search",
                  "icon": Object {
                    "default": Object {},
                    "grey": Object {},
                    "selected": Object {},
                  },
                  "name": "global-search",
                }
              }
              currentApp=""
            />
          </dt>
          <dt
            className="species-selectorIcon"
            key="species-selector"
          >
            <React.Memo
              app={
                Object {
                  "description": "species selector",
                  "icon": Object {
                    "default": Object {},
                    "grey": Object {},
                    "selected": Object {},
                  },
                  "name": "species-selector",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className="border"
          key="browsers"
        >
          <dt
            className="browserIcon"
            key="browser"
          >
            <React.Memo
              app={
                Object {
                  "description": "browser",
                  "icon": Object {
                    "default": Object {},
                    "selected": Object {},
                  },
                  "name": "browser",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className="border"
          key="tools"
        >
          <dt
            className="toolsIcon"
            key="tools"
          >
            <React.Memo
              app={
                Object {
                  "description": "tools",
                  "icon": Object {
                    "default": Object {},
                    "grey": Object {},
                    "selected": Object {},
                  },
                  "name": "tools",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className="border"
          key="downloads"
        >
          <dt
            className="downloadIcon"
            key="download"
          >
            <React.Memo
              app={
                Object {
                  "description": "download",
                  "icon": Object {
                    "default": Object {},
                    "grey": Object {},
                    "selected": Object {},
                  },
                  "name": "download",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
        <dl
          className=""
          key="learning"
        >
          <dt
            className="help-docsIcon"
            key="help-docs"
          >
            <React.Memo
              app={
                Object {
                  "description": "help and documentation",
                  "icon": Object {
                    "default": Object {},
                    "grey": Object {},
                    "selected": Object {},
                  },
                  "name": "help-docs",
                }
              }
              currentApp=""
            />
          </dt>
        </dl>
      </div>
    </div>
    <div
      className="about"
    >
      <dl>
        <dt
          className="aboutIcon"
        >
          <React.Memo
            app={
              Object {
                "description": "about ensembl",
                "icon": Object {
                  "default": Object {},
                  "selected": Object {},
                },
                "name": "about",
              }
            }
            currentApp=""
          />
        </dt>
      </dl>
    </div>
  </div>
</ForwardRef>
`;
